analysis/tbcshaman/src/modules/spells/totems/BaseTotem.tsx

```diff
@@ -0,0 +1,104 @@
+import { Trans } from '@lingui/macro';
+import { formatPercentage } from 'common/format';
+import { SpellIcon, SpellLink } from 'interface';
+import UptimeIcon from 'interface/icons/Uptime';
+import Analyzer, { Options } from 'parser/core/Analyzer';
+import { ThresholdStyle } from 'parser/core/ParseResults';
+import BoringValue from 'parser/ui/BoringValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+
+import { TotemElements, TOTEMS_BY_ELEMENT } from '../../../totemConstants';
+import TotemTracker from '../../features/TotemTracker';
+
+export default class BaseTotem extends Analyzer {
```

Comment:
not a required change but: thoughts on structuring as a single analyzer with multiple suggestion `get`ters + multiple statistics returned (using a fragment in `statistic`) rather than base class + a subclass per element?

I can't foresee a reason to vary the behavior here in a way that has the subclass be more than a fancy way of setting a parameter.

Reply:
I'm planning on adding element specific extras that didn't make sense to have in every totem element. 

Water totems should show mana/health restores
Fire totems should show how much damage they deal
Earth can show how much damage was reduced by stoneskin
Air can show spells grounded

I did the grounding totem as a one off, but I think I like the idea of having it as a part of the entire air totem package. I could go either way on that one.

Reply:
ah, that makes more sense. carry on :+1: